OpenRA.Mods.Common/Traits/Player/PlayerStatistics.cs

```diff
@@ -108,15 +108,24 @@ public void ResolveOrder(Actor self, Order order)
 	[Desc("Attach this to a unit to update observer stats.")]
 	public class UpdatesPlayerStatisticsInfo : TraitInfo<UpdatesPlayerStatistics> { }
 
-	public class UpdatesPlayerStatistics : INotifyKilled
+	public class UpdatesPlayerStatistics : INotifyKilled, INotifyCreated
 	{
+		PlayerStatistics defenderStats;
+		int cost;
+
+		void INotifyCreated.Created(Actor self)
+		{
+			defenderStats = self.Owner.PlayerActor.Trait<PlayerStatistics>();
```

Comment:
Does it matter if the unit changes its owner?

Reply:
It does, so we would need to account for that.

IMO it would be better to drop this commit.  These traits are only used once (or never), so this is moving the overhead from death to creation, which is just as bad wrt trait lookups!